### ISSUE TITLE
Why does VGGT predict normalized scale instead of metric scale like MAST3R?

### ISSUE THREAD

**Issue by Jian-Zhang-3DV** [facebookresearch/vggt#56]
I’ve been exploring VGGT and noticed that it predicts depth in a normalized scale rather than a metric scale, unlike MAST3R, which outputs metric depth directly. Could you share the reasoning behind this design choice? Is it related to generalization, training data, or a specific use case? I’d love to understand the motivation better. Thanks for your work on this!

**Comment by jytime** [comment]
Hi @Jian-Zhang-3DV ,

The main issue is the limited availability of metric data. In my early experiments, mixing metric and normalized data—especially for camera pose estimation—seemed to degrade performance. So I opted for a normalized prediction approach. We might revisit a metric-based model if sufficient metric data becomes available in the future.

**Comment by ThanosTsoukleidis-Karydakis** [comment]
I've been exploring VGGT as well, and I find it fascinating! I was wondering if you have any ideas or suggestions on how one might (approximately) convert the relative scale point map produced by VGGT into a metric scale? Specifically, I'm referring to the point map generated using the following code snippet:

`
point_map, point_conf = model.point_head(aggregated_tokens_list, images, ps_idx)
`

**Comment by jianglh-WHU** [comment]
I’m trying to understand how to correctly compute the `norm_factor` for normalizing the scene, as described in your paper.

> We follow [128] and, first, express all quantities in the coordinate frame of the first camera g1. Then, we compute the average Euclidean distance of all 3D points in the point map P to the origin and use this scale to normalize the camera translations t, the point map P , and the depth map D.

I followed the approach used in DUST3R to normalize the point map as follows:
```
dis = valid_pts.norm(dim=-1)
norm_factor = dis.mean()
```
However, I'm still observing a scale difference between the ground truth (GT) depth-unprojected point maps and the VGGT predicted maps.  Is that normal, or do you have any advice?

<img width="638" alt="Image" src="https://github.com/user-attachments/assets/c13313a6-1cf9-4fa6-a872-0dcae61c9d13" />

**Comment by jianglh-WHU** [comment]
BTW, could you tell the hyperparameter $\epsilon$ of huber loss which supervises camera poses? (1.0 or 0.1)

**Comment by mmmingxuan** [comment]
@jianglh-WHU  Hello, thank you for your exploration. I am very curious about how you visualized the results of the vggt above. Currently, I can only use the web-based viser, which is quite inconvenient. If you could share your code, I would be very grateful.

**Comment by vgutierrez2404** [comment]
You can take the model's prediction points and convert them into a ply format (this is added in the demo_colmap.py), and then use any 3d library like open3d or directly open it with any 3d visualizer.

**Comment by heMinger** [comment]
> I’m trying to understand how to correctly compute the `norm_factor` for normalizing the scene, as described in your paper.
> 
> > We follow [128] and, first, express all quantities in the coordinate frame of the first camera g1. Then, we compute the average Euclidean distance of all 3D points in the point map P to the origin and use this scale to normalize the camera translations t, the point map P , and the depth map D.
> 
> I followed the approach used in DUST3R to normalize the point map as follows:
> 
> ```
> dis = valid_pts.norm(dim=-1)
> norm_factor = dis.mean()
> ```
> 
> However, I'm still observing a scale difference between the GT depth-unprojected point maps and the VGGT predicted maps. Is that normal, or do you have any advice?
> 
> <img alt="Image" width="638" src="https://private-user-images.githubusercontent.com/71580859/427158393-c13313a6-1cf9-4fa6-a872-0dcae61c9d13.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjEwMzgwNjAsIm5iZiI6MTc2MTAzNzc2MCwicGF0aCI6Ii83MTU4MDg1OS80MjcxNTgzOTMtYzEzMzEzYTYtMWNmOS00ZmE2LWE4NzItMGRjYWU2MWM5ZDEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDIxVDA5MDkyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjYmNiZTdhYjgwZmNmNTM0ZTk3MGIyYjhiMDAyYzZkNGZmYmRlMTI2OTEzNGJkYTFiYWFkZTJkNjEyY2ZiYWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xnNx8CmEIH8H2gMcGaWeUYemDc5u_sxRKV_BphOvRcg">

Hello, I encountered the same mismatch between the predicted point clouds and the ground truth, even in a different dataset to yours. Have you found a way to fix this or match the two sets of points?
